src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -762,8 +814,30 @@ await _initLock.ExecuteNuGetOperationAsync(async () =>
                             {
                                 try
                                 {
-                                    var vsProjectAdapter = await _vsProjectAdapterProvider.CreateAdapterForFullyLoadedProjectAsync(hierarchy);
+                                    // TODO: Iterate these projects for project.json migration.
+                                    // Catch it - don't have cache.
+                                    //await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                                    //var projectSafeName = await project.GetCustomUniqueNameAsync();
+                                    //var nuGetProject = await GetNuGetProjectAsync(projectSafeName);
+                                    IVsProjectAdapter vsProjectAdapter = await _vsProjectAdapterProvider.CreateAdapterForFullyLoadedProjectAsync(hierarchy);
                                     await AddVsProjectAdapterToCacheAsync(vsProjectAdapter);
```

Comment:
already does a lot of what you'd be doing in the latter code, so think about moving it down.

Right now your approach is , add project, detect it's json, migrate and remove/re-add. 

I think doing it before we add is cleaner.